### PR TITLE
KATI-74[Settings] Doctor Settings UI

### PR DESCRIPTION
**Pull request by huuduy216**
@hieumai

### CODE REVIEW

File: app/src/main/java/com/kms/cura/controller/UserController.java [hunk 1]
```diff
@@ -13,6 +13,10 @@
 import com.kms.cura.entity.user.UserEntity;
 import com.kms.cura.model.UserModel;
 
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/controller/UserController.java]: Remove un-used imports

File: app/src/main/java/com/kms/cura/model/Settings.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
  */
 public class Settings {
     public static final String SERVER_URL = "http://192.168.74.141:8080/cura/cura_server";
+    public static final String SERVER_LOCAL_URL = "http://192.168.50.87:8080/com.kms.cura_server/cura_server";
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/Settings.java]: Remove local url string

File: app/src/main/java/com/kms/cura/model/UserModel.java [hunk 3]
```diff
@@ -17,6 +17,7 @@
 public class UserModel extends EntityModel {
     private static UserModel instance;
     private static String tag_string_req = "string_req";
+    private List<DoctorUserEntity> doctors = new ArrayList<>();
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/UserModel.java]: Remove this

File: app/src/main/java/com/kms/cura/utils/DoctorSearchResults.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.kms.cura.entity.user.DoctorUserEntity;
+import com.kms.cura.entity.user.UserEntity;
+
+import java.util.List;
+
+/**
+ * Created by duyhnguyen on 6/24/2016.
+ */
+public class DoctorSearchResults {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/utils/DoctorSearchResults.java]: This file is out of scope so remove too

File: app/src/main/java/com/kms/cura/view/activity/DoctorProfessionalSettingsActivity.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+    private void initExpandableListView() {
+        expListView = (ExpandableListView) findViewById(R.id.lvExp);
+
+        // preparing list data
+        prepareListData();
+
+        listAdapter = new ExpandableListAdapter(this, listDataHeader, listDataChild);
+        listAdapter.setChildButtonText(SPECIALTIES_WORKING_HOUR_POSITION, "Add Specialties");
+        listAdapter.setChildButtonText(FACILITIES_OTHER__POSITION, "Add Facilities");
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/DoctorProfessionalSettingsActivity.java]: "Add Specialties" and "Add Facilities" should be in resource file

File: app/src/main/java/com/kms/cura/view/activity/DoctorProfessionalSettingsActivity.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    /*
+     * Preparing the list data
+     */
+    private void prepareListData() {
+        listDataHeader = new ArrayList<String>();
+        listDataChild = new HashMap<String, List<String>>();
+
+        // Adding Group data
+        listDataHeader.add("Specialties");
+        listDataHeader.add("Facilities");
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/DoctorProfessionalSettingsActivity.java]: "Specialties" and "Facilities" should be in resource file too

File: app/src/main/java/com/kms/cura/view/adapter/ExpandableListAdapter.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    public void onClick(View v) {
+        if (v.getId() == R.id.button_list_item_button) {
+            Button button = (Button) v;
+            String buttonText = button.getText().toString();
+
+            if (buttonText.equals("Add Specialties")) {
+                Toast.makeText(v.getContext(), "Add new Specialities", Toast.LENGTH_SHORT).show();
+            } else if (buttonText.equals("Add Facilities")) {
+                Toast.makeText(v.getContext(), "Add new Facilities", Toast.LENGTH_SHORT).show();
+            }
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/adapter/ExpandableListAdapter.java]: After that then get "Add Specialties" and "Add Facilities" from resource file

File: app/src/main/res/layout/list_item_footer.xml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:paddingLeft="15sp"
+    android:paddingTop="15sp"
+    android:paddingBottom="15sp">
+
+    <Button
+        android:layout_width="@dimen/buttonWidth_150"
+        android:layout_height="wrap_content"
+        android:text="New Button"
```
Comment (hieumai) [inline app/src/main/res/layout/list_item_footer.xml]: "New Button" should be in resource file

**Comment by hieumai** [comment]
OK from me. @vtnlinh95  and @toannguyen71195  please start verifying this

**Comment by vtnlinh95** [comment]
@huuduy216  : 
+Why clicking on "other" does not show up anything ?
- The action for the UI to delete or add facilities and specialities will be implemented in this ticket or in the function ticket because the description for the fuction ticket does not contain that action
- Why don't you show the working hour, right now it is in the server and when log in we have the whole working hour ?
- The text in all the button should be white for the consistency

**Comment by huuduy216** [comment]
@vtnlinh95  
- The deletion will be implement in the function ticket
- fixed
- fixed

**Comment by hieumai** [comment]
@huuduy216 solve conflicts

**Comment by hieumai** [comment]
@huuduy216  rebase this so I can merge
